### DISCUSSION TITLE
Vue outside Rails?

### DISCUSSION THREAD

**Discussion by aemadrid-twx** [ElMassimo/vite_ruby#370] [Q&A]
Has anybody tried building a Vue app outside Rails? Like a straight up Vue app that the app code is transpiled from ruby to JS? All the info I'm seeing seems very straightforward from a running Rails/ruby app but not sure if vite_ruby can be used just as a vite plugin and how.

**Comment by ElMassimo** [comment]
Hi Adrian!

I'm not sure I understand the question, can you clarify and describe your use case?

**Comment by aemadrid-twx** [comment]
I would like to build a normal Vue app. But instead of using plain Javascript or Typescript I would like to use Ruby that RubyJS transpiles into Javascript through Vite Ruby. And this app would not be inside a Rails app. Is that possible/doable?

**Reply by ElMassimo** [reply]
I'm not familiar with RubyJS, but I guess that would require writing a Vite plugin that transpiles Ruby code in the Vue SFC.

For example, the plugin could use the Vue compiler to parse `<script lang="ruby">`, feed the content to RubyJS, and then rewrite the code (using the `transform` hook) and remove the `lang="ruby"` attribute.

In summary, it might be possible, and it sounds like an interesting thing to try out, but it would require creating a 
new Vite plugin.

**Comment by aemadrid-twx** [comment]
OK, it makes sense. I will take a look at what you are doing and see what I can learn from it. Thanks!

**Comment by aemadrid-twx** [comment]
And BTW I meant [Ruby2JS](https://www.ruby2js.com/) :-)
